docs/developer-docs/daos/nns/neuron-management.md

```diff
@@ -0,0 +1,272 @@
+Every user can stake ICP utility tokens into a neuron and pariticipate in the NNS DAO with this neuron.  
+
+* **Neuron ID**: Each neuron has an identity selected by NNS governance at neuron creation.
+* **Account**: Each neuron has an associated account on the ICP ledger where the locked ICP balance resides. This account is owned by the NNS governance canister and therefore a user cannot move staked tokens.
+* **Controller**: The principal that controls the neuron. The principal must identify a public key pair, which acts as a “master key,” such that the corresponding secret key should be kept very secure. A principal can control many neurons.
+
+* **Dissolve delay & dissolve state**: The tokens in a neuron are locked for a specified time, called the _dissolve delay_. This can be thought of like a kitchen timer that can only be turned in one direction. It can be arbitrarily increased, but only reduced by turning on the countdown and waiting for the time to pass. A neuron is eligible to vote if it has a dissolve delay of at least 6 months.
+  * A neuron can be _non-dissolving_ which means that the timer is stopped and the neuron's dissolve delay remains the same. 
+  * A neuron can be _dissolving_ which means that the timer is decreasing the neuron's dissolve delay as time passes. 
+  * Once the timer has counted down, a neuron is _dissolved_ and the ICP tokens can be unstaked again.
+* **Age**: Every non-dissolving neuron has an age, which denotes how long it has been in the non-dissolving state. 
+A neuron's dissolve delay and age increases the neuron's voting power. 
+
+* **Maturity**:  When neurons vote, over time the NNS increases their maturity to reward them. Maturity can be converted into ICP by spawning (see below).
+
+## Interacting with neurons
+All interactions with a neuron go through the following API on the NNS governance canister. This includes everything in a neuron's lifetime such as the neuron's creation, modifications and voting, and unstaking the ICP tokens.
+```
+type ManageNeuron = record {
+  id : opt NeuronId;
+  command : opt Command;
+  neuron_id_or_subaccount : opt NeuronIdOrSubaccount;
+};
+```
+
+### Creating a neuron and topping it up
+Staking ICP utility tokens in a neuron involves three steps:
+1. Compute the neuron's address on the ledger, where the staked tokens will be held. This corresponds to a subaccount of the NNS governance canister.
+2. Send ICP utility tokens to the neuron's address computed in (1).
+3. Claim the neuron on the NNS governance canister, which tells the governance canister that the transfer in (2) happened, upon which the NNS governance will create a new neuron.
+
+Let us go through these steps in more detail.
+
+**Compute the neuron's address**
+
+To compute the neuron's address in the first Step, proceed as follows.
+
+a) Learn the principal that should control the neuron and choose a nonce.
```

Comment:
```suggestion
a) Learn the principal that should control the neuron and choose a nonce. (The nonce is used to allow the same principal to control multiple neurons and does not need to be kept secret.)
```
BTW, is it called a nonce or a memo?

Reply:
I think the problem is that we are inconsistent. I would say "nonce" is clearer (a number only used once is what we want here), but the ledger field that defines this for neurons is called "memo" - the memo field is more general than this, but I think for this reason we eneded up calling it "memo" in some places